### PR TITLE
Edits to Optimistic UI article

### PR DESCRIPTION
**Pull request by StephenBarlow**
This article's been getting a fair amount of feedback lately so figured I'd take a pass at it

### CODE REVIEW

File: docs/source/performance/optimistic-ui.mdx [hunk 1]
```diff
@@ -1,20 +1,32 @@
 ---
-title: Optimistic UI
+title: Optimistic mutation results
+description: Update your UI before your server responds
 ---
 
-Optimistic UI is a pattern that you can use to simulate the results of a mutation and update the UI even before receiving a response from the server. Once the response is received from the server, the optimistic result is thrown away and replaced with the actual result.
+It's often possible to predict a mutation operation's most likely result _before_ your GraphQL server returns it. Apollo Client can use this "most likely result" to update your UI **optimistically**, making your app feel more responsive to the user.
```
Comment (brainkim) [inline docs/source/performance/optimistic-ui.mdx:6]: I tripped up on this first sentence the first time I read it. Maybe:
It's often possible to predict the most likely result of a mutation _before_ your GraphQL server returns it.

File: docs/source/performance/optimistic-ui.mdx [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
+Our app knows what the updated `Comment` object will probably look like, which means it can optimistically update its UI to display the update _before_ the GraphQL server responds with it. If our app is wrong (e.g., the GraphQL server returns an _unchanged_ `Comment` due to an error), the UI will automatically update to reflect the _actual_ response.
 
-The main way to get GraphQL data into your UI components with Apollo is to use a query, so if we want our optimistic response to update the UI, we have to make sure to return an optimistic response that will update the correct query result. Learn more about how to do this with the [`dataIdFromObject`](../caching/cache-configuration/#data-normalization) option.
+## The `optimisticResponse` option
 
-Here's what this looks like in the code:
+To enable this optimistic UI behavior, we provide an `optimisticResponse` option to the [mutate function](../data/mutations/#calling-the-mutate-function) we use to execute our mutation.
```
Comment (brainkim) [inline docs/source/performance/optimistic-ui.mdx:24]: , which we use

Maybe?

**Review by brainkim — APPROVED** [review]
This is dope as heck!

**Review by benjamn — COMMENTED** [review]
@StephenBarlow Echoing @brainkim, this is a great edit! Thanks!

**Pull request merged**